Not all artifacts saved during javac crash

For the following particular job:
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108/console

With the following "Rerun in Grinder" link:
https://ci.eclipse.org/openj9/job/Grinder/parambuild/?TARGET=-f+parallelList.mk+testList_0&SDK_RESOURCE=upstream&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=&VENDOR_TEST_DIRS=&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&RUNTIME_NAME=testList_0&PLATFORM=x86-64_linux&KEEP_REPORTDIR=false&PERSONAL_BUILD=false&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2FAdoptOpenJDK%2Fopenjdk-tests.git&LABEL=&EXTRA_OPTIONS=&BUILD_IDENTIFIER=eclipse%2Fopenj9%2311393&CUSTOMIZED_SDK_URL=https%3A%2F%2F140-211-168-230-openstack.osuosl.org%2Fartifactory%2Fci-eclipse-openj9%2FBuild_JDK8_x86-64_linux_Personal%2F982%2FOpenJ9-JDK8-x86-64_linux-20210104-160146.tar.gz+https%3A%2F%2F140-211-168-230-openstack.osuosl.org%2Fartifactory%2Fci-eclipse-openj9%2FBuild_JDK8_x86-64_linux_Personal%2F982%2Ftest-images.tar.gz&ADOPTOPENJDK_BRANCH=master&ARTIFACTORY_SERVER=ci-eclipse-openj9&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=&JDK_VERSION=8&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk8.git&OPENJ9_BRANCH=master&OPENJ9_SHA=origin%2Fpr%2F11393%2Fmerge&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse%2Fopenj9.git&UPSTREAM_JOB_NAME=&CUSTOM_TARGET=&VENDOR_TEST_SHAS=&JDK_BRANCH=openj9&LABEL_ADDITION=&ARTIFACTORY_REPO=ci-eclipse-openj9&ARTIFACTORY_ROOT_DIR=Test&UPSTREAM_TEST_JOB_NUMBER=728&DOCKERIMAGE_TAG=&JDK_IMPL=openj9&SSH_AGENT_CREDENTIAL=&AUTO_DETECT=true&TKG_SHA=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=ab89294b-5ba1-48e9-8c85-107daca5a2e9&NUM_MACHINES=&OPENJDK_SHA=&BUILD_LIST=functional&UPSTREAM_JOB_NUMBER=&TIME_LIMIT=10&JVM_OPTIONS=&PARALLEL=None

It appears only the system core dump is saved in the artifacts. However from stdout we can see the javacore, jitdump, and Snap trace are all generated.

```
18:07:27  compile:
18:07:27       [echo] Ant version is Apache Ant(TM) version 1.10.5 compiled on July 10 2018
18:07:27       [echo] ============COMPILER SETTINGS============
18:07:27       [echo] ===fork:                         yes
18:07:27       [echo] ===executable:                   /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdkbinary/j2sdk-image/bin/javac
18:07:27       [echo] ===debug:                        on
18:07:27       [echo] ===destdir:                      /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/TKG/../../jvmtest/functional/IllegalAccessError_for_protected_method
18:07:27      [javac] Compiling 2 source files to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/bin
18:07:29      [javac] Unhandled exception
18:07:29      [javac] Type=Segmentation error vmState=0x00000000
18:07:29      [javac] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
18:07:29      [javac] Handler1=00007F92D5DC1DA0 Handler2=00007F92D56A2020 InaccessibleAddress=00000000FFB80418
18:07:29      [javac] RDI=0000000000048D00 RSI=0000000000048D00 RAX=0000000000048D00 RBX=00000000FFB80400
18:07:29      [javac] RCX=00000000FFB80400 RDX=00000000020E0001 R8=0000000000000000 R9=00000007FFB42360
18:07:29      [javac] R10=0000000000051300 R11=00007F92D73B8150 R12=00000000000E3558 R13=0000000000000001
18:07:29      [javac] R14=00007F927C6680D3 R15=00007F92D6D02910
18:07:29      [javac] RIP=00007F92D5D4D763 GS=0000 FS=0000 RSP=00007F92D6D02590
18:07:29      [javac] EFlags=0000000000010246 CS=0033 RBP=0000000000000000 ERR=0000000000000004
18:07:29      [javac] TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=00000000FFB80418
18:07:29      [javac] xmm0 0000000000000005 (f: 5.000000, d: 2.470328e-323)
18:07:29      [javac] xmm1 cc31990665dbbe9e (f: 1708899968.000000, d: -1.104629e+59)
18:07:29      [javac] xmm2 00007f92d6d02940 (f: 3603966208.000000, d: 6.930192e-310)
18:07:29      [javac] xmm3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
18:07:29      [javac] xmm4 43e0000000000000 (f: 0.000000, d: 9.223372e+18)
18:07:29      [javac] xmm5 000000003d9bcb21 (f: 1033620288.000000, d: 5.106763e-315)
18:07:29      [javac] xmm6 000000004928d030 (f: 1227411456.000000, d: 6.064219e-315)
18:07:29      [javac] xmm7 00000000be871679 (f: 3196524032.000000, d: 1.579293e-314)
18:07:29      [javac] xmm8 410440d286af43d9 (f: 2259633152.000000, d: 1.659143e+05)
18:07:29      [javac] xmm9 43e0000000000000 (f: 0.000000, d: 9.223372e+18)
18:07:29      [javac] xmm10 3f847ae147ae147b (f: 1202590848.000000, d: 1.000000e-02)
18:07:29      [javac] xmm11 3fefae147ae147ae (f: 2061584256.000000, d: 9.900000e-01)
18:07:29      [javac] xmm12 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
18:07:29      [javac] xmm13 4107ffc000000000 (f: 0.000000, d: 1.966000e+05)
18:07:29      [javac] xmm14 0000000048473240 (f: 1212625408.000000, d: 5.991166e-315)
18:07:29      [javac] xmm15 00000000469f5b68 (f: 1184848768.000000, d: 5.853931e-315)
18:07:29      [javac] Module=/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdkbinary/j2sdk-image/jre/lib/amd64/compressedrefs/libj9vm29.so
18:07:29      [javac] Module_base_address=00007F92D5D28000
18:07:29      [javac] Target=2_90_20210104_982 (Linux 4.4.0-169-generic)
18:07:29      [javac] CPU=amd64 (4 logical CPUs) (0x5e2f07000 RAM)
18:07:29      [javac] ----------- Stack Backtrace -----------
18:07:29      [javac] (0x00007F92D5D4D763 [libj9vm29.so+0x25763])
18:07:29      [javac] (0x00007F92D5D3AA50 [libj9vm29.so+0x12a50])
18:07:29      [javac] (0x00007F92D5DFE7A2 [libj9vm29.so+0xd67a2])
18:07:29      [javac] ---------------------------------------
18:07:29      [javac] JVMDUMP039I Processing dump event "gpf", detail "" at 2021/01/04 19:07:28 - please wait.
18:07:29      [javac] JVMDUMP032I JVM requested System dump using '/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/core.20210104.190728.11521.0001.dmp' in response to an event
18:07:34      [javac] JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.11546.
18:07:34      [javac] 
18:07:34      [javac] JVMDUMP010I System dump written to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/core.20210104.190728.11521.0001.dmp
18:07:34      [javac] JVMDUMP032I JVM requested Java dump using '/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/javacore.20210104.190728.11521.0002.txt' in response to an event
18:07:34      [javac] JVMDUMP010I Java dump written to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/javacore.20210104.190728.11521.0002.txt
18:07:34      [javac] JVMDUMP032I JVM requested Snap dump using '/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/Snap.20210104.190728.11521.0003.trc' in response to an event
18:07:34      [javac] JVMDUMP010I Snap dump written to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/Snap.20210104.190728.11521.0003.trc
18:07:34      [javac] JVMDUMP032I JVM requested JIT dump using '/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/jitdump.20210104.190728.11521.0004.dmp' in response to an event
18:07:34      [javac] JVMDUMP010I JIT dump written to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/jitdump.20210104.190728.11521.0004.dmp
18:07:34      [javac] JVMDUMP013I Processed dump event "gpf", detail "".
18:07:34  
18:07:34  BUILD FAILED
18:07:34  /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/TKG/scripts/build_test.xml:60: The following error occurred while executing this line:
18:07:34  /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/build.xml:52: The following error occurred while executing this line:
18:07:34  /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/build.xml:81: The following error occurred while executing this line:
18:07:34  /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/functional/IllegalAccessError_for_protected_method/build.xml:54: Compile failed; see the compiler error output for details.
18:07:34  
18:07:34  Total time: 18 seconds
18:07:34  1
18:07:34  20210104-19:07:33 - Successfully moved core file core.20210104.190728.11521.0001.dmp to /home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/TKG/../TKG/test_output_compilation
18:07:34  compile.mk:32: recipe for target 'compile' failed
18:07:34  make[1]: *** [compile] Error 1
18:07:34  make[1]: Leaving directory '/home/****/workspace/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/openjdk-tests/TKG'
18:07:34  makefile:59: recipe for target 'compile' failed
18:07:34  make: *** [compile] Error 2
[Pipeline] }
18:07:35  $ ssh-agent -k
18:07:35  unset SSH_AUTH_SOCK;
18:07:35  unset SSH_AGENT_PID;
18:07:35  echo Agent pid 11173 killed;
18:07:36  [ssh-agent] Stopped.
[Pipeline] // sshagent
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Post)
[Pipeline] timestamps
[Pipeline] {
[Pipeline] step
18:07:37  TAP Reports Processing: START
18:07:37  Looking for TAP results report in workspace using pattern: **/*.tap
18:07:46  Did not find any matching files. Setting build result to FAILURE.
[Pipeline] archiveArtifacts
18:07:46  Archiving artifacts
[Pipeline] junit
18:07:49  Recording test results
[Pipeline] getArtifactoryServer
[Pipeline] newBuildInfo
[Pipeline] artifactoryUpload
18:09:00  [consumer_0] Deploying artifact: https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Test/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108/SHA.txt
[Pipeline] publishBuildInfo
18:09:47  Deploying build info to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
18:09:47  Deploying build descriptor to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
18:09:48  Build successfully deployed. Browse it in Artifactory under https://140-211-168-230-openstack.osuosl.org/artifactory/webapp/builds/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108
18:09:48  Sending request for build retention, deleting build artifacts, max number of builds to store: 20.
[Pipeline] echo
18:09:48  Test output artifactory URL:'https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Test/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108/'
[Pipeline] sh
18:09:51  + tar -cf - ./openjdk-tests/TKG/test_output_compilation
18:09:51  + pigz -9
[Pipeline] getArtifactoryServer
[Pipeline] newBuildInfo
[Pipeline] artifactoryUpload
18:09:59  [consumer_0] Deploying artifact: https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Test/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108/functional_test_output.tar.gz
[Pipeline] publishBuildInfo
18:10:54  Deploying build info to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
18:10:55  Deploying build descriptor to: https://140-211-168-230-openstack.osuosl.org/artifactory/api/build
18:10:55  Build successfully deployed. Browse it in Artifactory under https://140-211-168-230-openstack.osuosl.org/artifactory/webapp/builds/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108
18:10:55  Sending request for build retention, deleting build artifacts, max number of builds to store: 20.
[Pipeline] echo
18:10:55  Test output artifactory URL:'https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Test/Test_openjdk8_j9_sanity.functional_x86-64_linux_Personal_testList_0/108/'
[Pipeline] }
[Pipeline] // timestamps
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
terminateJavaProcesses iteration: 1
[Pipeline] echo
PROCESSCATCH: Checking for any leftover java processes on the machine
[Pipeline] sh
+ ps -fu jenkins
+ grep java
+ egrep -v remoting.jar|agent.jar|grep
jenkins  10104 10103  0 15:30 ?        00:01:56 java -jar slave.jar
[Pipeline] echo
PROCESSCATCH: There are rogue processes still on the machine as per the list above
[Pipeline] echo
PROCESSCATCH: Removed processes - here is what is remaining:
[Pipeline] sh
+ ps -fu jenkins
+ egrep+  -v remoting.jar|agent.jar|grep
grep java
jenkins  10104 10103  0 15:30 ?        00:01:56 java -jar slave.jar
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is disabled by the job configuration...
[WS-CLEANUP] done
[Pipeline] sh
+ find /tmp -name *core* -print -exec rm -f {} ;
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // stage
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE
```

We should fix this so all artifacts are saved.

--- a/scripts/moveDmp.pl
+++ b/scripts/moveDmp.pl
@@ -125,58 +125,70 @@ sub checkLog {
 
 sub moveTDUMPS {
 	my ($file, $moveLocation) = @_;
-	my @dumplist = ();
-	# Use a hash to ensure that each dump is only dealt with once
+	if(!$file) {return;}
+
 	my %parsedNames = ();
-	if ($^O ne 'os390') {
-		if($file) {
-			while ($file =~ /System dump written to (.*)/g) {
-				my $curCorePath = $1;
-				$curCorePath =~ s/\r//g;
-				my $curCoreAbsPath = "";
-				my $moveLocationAbs = "";
-				if($^O =~ /cygwin/) {
-					$curCoreAbsPath = qx(cygpath -u '$curCorePath');
-					$moveLocationAbs = $moveLocation;
-					$moveLocation = qx(cygpath -w '$moveLocation');
+	
+	my $dumpType = qr/(System|Java|Snap|JIT|Heap)/;
+	if ($^O eq 'os390') {
+		# Use different logic to handle zos system dump
+		$dumpType = qr/(Java|Snap|JIT|Heap)/;
+	}
+	while ($file =~ /${dumpType} dump written to (.*)/g) {
+		my $curCorePath = $2;
+		$curCorePath =~ s/\r//g;
+		my $curCoreAbsPath = "";
+		my $moveLocationAbs = "";
+		if($^O =~ /cygwin/) {
+			$curCoreAbsPath = qx(cygpath -u '$curCorePath');
+			$moveLocationAbs = $moveLocation;
+			$moveLocation = qx(cygpath -w '$moveLocation');
+		} else {
+			$curCoreAbsPath = Cwd::abs_path( $curCorePath );
+			$moveLocationAbs = Cwd::abs_path( $moveLocation );
+		}
+		my $curCoreName = basename($curCoreAbsPath);
+		if(!exists $parsedNames{$curCoreName}) {
+			# handle each core only once
+			$parsedNames{$curCoreName} = 1;
+			# If the core is not in the preferred location, move it to $moveLocation
+			if($curCoreAbsPath !~  $moveLocationAbs) {
+				qx(mv '${curCorePath}' ${moveLocationAbs});
+				my $moveResult = $?;
+				$moveResult = $moveResult >> 8 unless ($moveResult == -1);
+				if($moveResult == 0) {
+					logMsg("Successfully moved core file $curCoreName to $moveLocation");
 				} else {
-					$curCoreAbsPath = Cwd::abs_path( $curCorePath );
-					$moveLocationAbs = Cwd::abs_path( $moveLocation );
-				}
-				my $curCoreName = basename($curCoreAbsPath);
-				if(!exists $parsedNames{$curCoreName}) {
-					# handle each core only once
-					$parsedNames{$curCoreName} = 1;
-					# If the core is not in the preferred location, move it to $moveLocation
-					if($curCoreAbsPath !~  $moveLocationAbs) {
-						qx(mv '${curCorePath}' ${moveLocationAbs});
-						my $moveResult = $?;
-						$moveResult = $moveResult >> 8 unless ($moveResult == -1);
-						if($moveResult == 0) {
-							logMsg("Successfully moved core file $curCoreName to $moveLocation");
-						} else {
-							logMsg("Failed to move core file from $curCorePath to $moveLocation");
-						}
-					}
+					logMsg("Failed to move core file from $curCorePath to $moveLocation");
 				}
 			}
 		}
-		return;
 	}
+
+	if ($^O eq 'os390') {
+		moveSysDumpforZOS($file, $moveLocation);
+	}
+}
+
+sub moveSysDumpforZOS {
+	my ($file, $moveLocation) = @_;
+
+	my @dumplist = ();
+	# Use a hash to ensure that each dump is only dealt with once
+	my %parsedNames = ();
 	
-	if ($file) {
-		while ($file =~ /IEATDUMP success for DSN='(.*)'/g) {
-			$parsedNames{$1} = 1;
-		}
-		# A partial dump has been created, even though it is a failure, dump still occurred and needs moving
-		while ($file =~ /IEATDUMP failure for DSN='(.*)' RC=0x00000004 RSN=0x00000000/g) {
-			$parsedNames{$1} = 1;
-		}
-		# Dump failed due to no space left on the machine, so print out warning message
-		while ($file =~ /IEATDUMP failure for DSN='(.*)' RC=0x00000008 RSN=0x00000026/g) {
-			logMsg("ERROR: TDUMP failed due to no space left on machine. $1 cannot be found.");
-		}
+	while ($file =~ /IEATDUMP success for DSN='(.*)'/g) {
+		$parsedNames{$1} = 1;
 	}
+	# A partial dump has been created, even though it is a failure, dump still occurred and needs moving
+	while ($file =~ /IEATDUMP failure for DSN='(.*)' RC=0x00000004 RSN=0x00000000/g) {
+		$parsedNames{$1} = 1;
+	}
+	# Dump failed due to no space left on the machine, so print out warning message
+	while ($file =~ /IEATDUMP failure for DSN='(.*)' RC=0x00000008 RSN=0x00000026/g) {
+		logMsg("ERROR: TDUMP failed due to no space left on machine. $1 cannot be found.");
+	}
+
 	push(@dumplist, keys(%parsedNames));
 	if (!@dumplist) {
 		# Nothing to do

